forLim.py

```diff
@@ -535,517 +548,552 @@ def run(self):
+                        merge_fct.main(files,merge_path,args)
+                        
+                        
+                    #Print progress on user window
+                    self.dlg.progressBar.setValue(self.dlg.progressBar.value()+1)
+                    QApplication.processEvents() 
                     
-                    #Get projection reference
-                    ds = ogr.Open(lastPath[0])
-                    lyr = ds.GetLayer()
-                    proj = lyr.GetSpatialRef()
-                    ds.Destroy()
                     
-                    #Create temporary file containing polygons of extents
-                    tmp_ds = driver.CreateDataSource(tmp_path)
-                    tmp_lyr = tmp_ds.CreateLayer('mask',geom_type=ogr.wkbPolygon,srs=proj)
                     
+                    ###############################
+                    #  2. dissoudre les éléments  #
+                    ###############################
                     
-                    files = os.listdir(args["Path_output_tiles"])
-                    for f in files:
-                        if f.endswith(".shp"):
+                    if nfiles >= 1:
+                        #Print progress on user window
+                        self.dlg.label_printActualProcess.setText("Dissolving merged shapefile ...")
+                        QApplication.processEvents()
+                        
+                        
+                        #Dissolve on FD and PB
+                        file_type = ["FD", "PBandFD"]
+                        for i in file_type:
                             
-                            #Get extent
-                            ds = driver.Open(os.path.join(args["Path_output_tiles"],f))
-                            lyr = ds.GetLayer()
-                            xmin,xmax,ymin,ymax = lyr.GetExtent()
+                            #Create new files
+                            merge_path = os.path.join(args["Path_output_merge"], "merge_" + i + ".shp")
+                            dissolve_path = os.path.join(args["Path_output_merge"], "dissolve_" + i + ".shp")
                             
-                            #Write new polygon
-                            #Create ring
-                            ring = ogr.Geometry(ogr.wkbLinearRing)
-                            ring.AddPoint(xmin,ymin)
-                            ring.AddPoint(xmin,ymax)
-                            ring.AddPoint(xmax,ymax)
-                            ring.AddPoint(xmax,ymin)
-                            ring.AddPoint(xmin,ymin)
                             
-                            #Create polygon
-                            poly = ogr.Geometry(ogr.wkbPolygon)
-                            poly.AddGeometry(ring)
+                            #Dissolve merge files
+                            runalg("gdalogr:dissolvepolygons", merge_path, "geometry", "TYPE", True, False, False, False, False, "diss", [], dissolve_path)
```

Comment:
Maybe you should try to run everything in RAM and make use of QGIS memory layers. But, you'll have to check that this is reasonable on standard computer with something like 8Go RAM...